### PR TITLE
Enhanced staking contracts

### PR DESCRIPTION
**Pull request by frostornge**
* additional enumerative query functions
* DAO DAO v1 compatible query messages

### CODE REVIEW

File: contracts/dao/src/helpers.rs [hunk 1]
```diff
@@ -61,14 +61,14 @@ pub fn get_voting_power_at_height(
     height: u64,
 ) -> StdResult<Uint128> {
     // Get voting power at height
-    let balance: ion_stake::msg::StakedBalanceAtHeightResponse = querier.query_wasm_smart(
+    let balance: ion_stake::msg::VotingPowerAtHeightResponse = querier.query_wasm_smart(
         staking_contract,
-        &ion_stake::msg::QueryMsg::StakedBalanceAtHeight {
+        &ion_stake::msg::QueryMsg::VotingPowerAtHeight {
             address: address.to_string(),
```
Comment (ezekiiel) [inline contracts/dao/src/helpers.rs:67]:
```suggestion
            address: address.into_string(),
```

`to_string` does a heap allocation whereas `into_string` will just reuse the allocated memory if possible.

File: contracts/stake/src/contract.rs [hunk 2]
```diff
@@ -318,6 +337,56 @@ pub fn query_total_value(deps: Deps, _env: Env) -> StdResult<TotalValueResponse>
...
+                )),
+                false => Ok((
+                    staker,
+                    StakedValueResponse {
+                        value: staked
+                            .checked_mul(balance)
+                            .map_err(StdError::overflow)?
+                            .checked_div(total)
+                            .map_err(StdError::divide_by_zero)?,
+                    },
```
Comment (ezekiiel) [inline contracts/stake/src/contract.rs:381]: Might be a nice idea to merge this zero check and multiplication with the one in `query_staked_value` as it's using the same logic.

File: contracts/stake/src/msg.rs [hunk 3]
```diff
@@ -26,37 +26,63 @@ pub enum ExecuteMsg {
...
+
+    RangeStakers {
+        start_at: Option<String>,
+        limit: Option<u32>,
+        order: Option<RangeOrder>,
+    },
+
+    GetConfig {},
+    Info {},
+    Dao {},
```
Comment (ezekiiel) [inline contracts/stake/src/msg.rs:72]: Awesome to add these!

**Review by ezekiiel — APPROVED** [review]

File: contracts/stake/src/contract.rs (hunk 2 above)
Comment (frostornge) [reply contracts/stake/src/contract.rs:381]: I agree, i'll add it new helper method for this

**Pull request merged**